server/src/main/java/org/apache/druid/client/DirectDruidClient.java

```diff
@@ -465,38 +469,49 @@ public void onFailure(Throwable t)
             {
               openConnections.getAndDecrement();
               if (future.isCancelled()) {
-                // forward the cancellation to underlying queriable node
+                cancel();
+              }
+            }
+
+            private void cancel()
```

Comment:
I'd call this method a little more specific: `cancelQuery()` and move up: make it a method of `DirectDruidClient`, to reduce internal indentation/nesting.